nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/json/JsonRecordSource.java

```diff
@@ -19,37 +19,59 @@
 import com.fasterxml.jackson.core.JsonFactory;
 import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.io.SerializedString;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.nifi.schema.inference.RecordSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.io.InputStream;
 
 public class JsonRecordSource implements RecordSource<JsonNode> {
+    private static final Logger logger = LoggerFactory.getLogger(JsonRecordSource.class);
     private static final JsonFactory jsonFactory;
     private final JsonParser jsonParser;
+    private final String skipToNestedJsonField;
 
     static {
         jsonFactory = new JsonFactory();
         jsonFactory.setCodec(new ObjectMapper());
     }
 
-    public JsonRecordSource(final InputStream in) throws IOException {
-        jsonParser = jsonFactory.createJsonParser(in);
+    public JsonRecordSource(final InputStream in, final String skipToNestedJsonField) throws IOException {
```

Comment:
Similar to other usage, recommend renaming the parameter:
```suggestion
    public JsonRecordSource(final InputStream in, final String nestedFieldName) throws IOException {
```